inc/Editor/BlockManagement/ConfigRegistry.php

```diff
@@ -78,14 +76,33 @@ public static function register_block( array $user_config = [] ): bool|WP_Error
 							'type' => $input_var['type'] ?? 'string',
 						];
 					}, array_keys( $input_schema ), array_values( $input_schema ) ),
-					'name' => $is_loop ? 'Collection' : 'Manual input',
+					'name' => 'Manual input',
 					'query_key' => self::DISPLAY_QUERY_KEY,
-					'type' => $is_loop ? 'loop' : 'input',
+					'type' => 'input',
 				],
 			],
 			'title' => $block_title,
 		];
 
+		// If the render query has no required input variables and outputs a
+		// collection, we should provide a "load collection" selector.
+		$required_variables = array_filter( $input_schema, function ( array $input_var ): bool {
+			return true === ( $input_var['required'] ?? true );
+		} );
+		if ( empty( $required_variables ) && true === ( $output_schema['is_collection'] ?? false ) ) {
+			// Add a "load collection" selector.
+			array_unshift(
+				$config['selectors'],
+				[
+					'image_url' => $display_query->get_image_url(),
+					'inputs' => [],
+					'name' => 'Load collection',
+					'query_key' => self::DISPLAY_QUERY_KEY,
+					'type' => 'load-collection',
+				]
+			);
+		}
+
```

Comment:
I think instead of adding another selector, we should add a conditional check to the single selector to see if it's type `input` or `load-collection` (can we rename it to `collection`)? So it behaves the same as before. Otherwise, we will get the 'Load collection' button and 'Provide manual input' buttons, like this: 

An example of a collection with no inputs: 
<img width="977" alt="Screenshot 2025-03-25 at 2 47 31 PM" src="https://github.com/user-attachments/assets/53b8b719-64c9-4ee2-bd71-136ef45b9326" />

And the built-in Airtable loop block:
<img width="968" alt="Screenshot 2025-03-25 at 3 29 01 PM" src="https://github.com/user-attachments/assets/35698382-dbb0-429a-a41f-ac87dfd44e03" />

Reply:
I borrowed the logic here for a related fix: https://github.com/Automattic/remote-data-blocks/pull/435

But I added it as a conditional in the `render_query` as I mentioned above. But if there's a reason it should be a selector, we can use the version here instead!